src/main/java/com/example/bookstoreapp/repository/impl/BookRepositoryImpl.java

```diff
@@ -0,0 +1,53 @@
+package com.example.bookstoreapp.repository.impl;
+
+import com.example.bookstoreapp.entity.Book;
+import com.example.bookstoreapp.repository.BookRepository;
+import jakarta.persistence.EntityManager;
+import jakarta.persistence.EntityManagerFactory;
+import jakarta.persistence.EntityTransaction;
+import jakarta.persistence.criteria.CriteriaBuilder;
+import jakarta.persistence.criteria.CriteriaQuery;
+import jakarta.persistence.criteria.Root;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class BookRepositoryImpl implements BookRepository {
+
+    private static final String CAN_T_SAVE_BOOK_MSG = "Can't save book";
+    private final EntityManagerFactory entityManagerFactory;
+
+    @Autowired
+    public BookRepositoryImpl(EntityManagerFactory entityManagerFactory) {
+        this.entityManagerFactory = entityManagerFactory;
+    }
+
+    @Override
+    public Book save(Book book) {
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        EntityTransaction transaction = entityManager.getTransaction();
+        try {
+            transaction.begin();
+            entityManager.persist(book);
+            transaction.commit();
+        } catch (Exception e) {
+            if (transaction.isActive()) {
+                transaction.rollback();
+            }
+            throw new RuntimeException(CAN_T_SAVE_BOOK_MSG + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public List<Book> findAll() {
+        EntityManager entityManager = entityManagerFactory.createEntityManager();
+        entityManager.createQuery("FROM Book");
+        CriteriaBuilder criteriaBuilder = entityManager.getCriteriaBuilder();
+        CriteriaQuery<Book> criteriaQuery = criteriaBuilder.createQuery(Book.class);
+        Root<Book> root = criteriaQuery.from(Book.class);
+        criteriaQuery.select(root);
+        return entityManager.createQuery(criteriaQuery).getResultList();
```

Comment:
You can simplify it 
```
        try (Session session = sessionFactory.openSession()) {
            return session.createQuery("FROM Book", Book.class).getResultList();
        } catch (Exception e) {
            throw new DataProcessingException("Can't get all books from DB", e);
        }
```

Reply:
Why do you recommend using Session when it's better to use EntityManager moreover I have already used it in my code?